test/UtilsTest.php

```diff
@@ -262,43 +267,51 @@ public function testCheckModuleInstallation()
 				]],
 			]);
 
-		// success
+		return $oRestClient;
+	}
+
+	public function testCheckModuleInstallation_ModuleFound()
+	{
+		$oRestClient = $this->PrepareCheckModuleInstallation();
 		$this->assertTrue(Utils::CheckModuleInstallation('itop-structure', true, $oRestClient));
-		$this->assertTrue(Utils::CheckModuleInstallation('itop-structure/0.0.0', false, $oRestClient));
-		$this->assertFalse(Utils::CheckModuleInstallation('itop-structure/1.2.3', false, $oRestClient));
+		$this->assertTrue(Utils::CheckModuleInstallation('itop-structure', true, $oRestClient));
+	}
 
-		// fail
-		$this->assertFalse(Utils::CheckModuleInstallation('fake-module', false, $oRestClient));
-		$this->expectExceptionMessage('Required iTop module fake-module is considered as not installed due to: Found: 0');
-		Utils::CheckModuleInstallation('fake-module', true, $oRestClient);
+	public function testCheckModuleInstallation_ModuleVersionFound()
+	{
+		$oRestClient = $this->PrepareCheckModuleInstallation();
+		$this->assertTrue(Utils::CheckModuleInstallation('itop-structure/1.0.0', true, $oRestClient));
+		$this->assertTrue(Utils::CheckModuleInstallation('itop-structure/1.0.0', true, $oRestClient));
 	}
 
-	public function testGetModuleVersion(): void
+	public function testCheckModuleInstallation_ModuleVersionNotFound()
 	{
-		$oRestClient = $this->createMock(\RestClient::class);
+		$oRestClient = $this->PrepareCheckModuleInstallation();
+		$this->assertFalse(Utils::CheckModuleInstallation('itop-structure/1.2.3', false, $oRestClient));
 
-		$oReflectionLastInstallDate = new \ReflectionProperty(Utils::class, 'sLastInstallDate');
-		$oReflectionLastInstallDate->setValue(null, '0000-00-00 00:00:00');
+		$this->expectExceptionMessage('Required iTop module itop-structure is considered as not installed due to: Version mismatch (1.0.0 >= 1.2.3)');
+		Utils::CheckModuleInstallation('itop-structure/1.2.3', true, $oRestClient);
+	}
 
-		$oRestClient->expects($this->atMost(2))
-			->method('Get')
-			->willReturnMap([
-				['ModuleInstallation', ['name' => 'itop-structure', 'installed' => '0000-00-00 00:00:00'], 'version', 1, [
-					'code' => 0,
-					'objects' => ['ModuleInstallation::0' => ['fields' => ['version' => '0.0.0']]],
-					'message' => 'Found: 1',
-				]],
-				['ModuleInstallation', ['name' => 'fake-module', 'installed' => '0000-00-00 00:00:00'], 'version', 1, [
-					'code' => 0,
-					'objects' => null,
-					'message' => 'Found: 0',
-				]],
-			]);
+	public function testCheckModuleInstallation_ModuleNotFound()
+	{
+		$oRestClient = $this->PrepareCheckModuleInstallation(2);
+		$this->assertFalse(Utils::CheckModuleInstallation('fake-module', false, $oRestClient));
+
+		$this->expectExceptionMessage('Required iTop module fake-module is considered as not installed due to: Found: 0');
+		$this->assertFalse(Utils::CheckModuleInstallation('fake-module', true, $oRestClient));
```

Comment:
Will this assert succeed given it will throw an exception?

Reply:
my guess is that 'assertFalse' is useless. it will not fail as i will not be reached.

Reply:
My idea as well, so it could have been replaced with this:
```suggestion
		Utils::CheckModuleInstallation('fake-module', true, $oRestClient);
```